browser/page_mapping.go

```diff
@@ -259,6 +256,25 @@ func mapPage(vu moduleVU, p *common.Page) mapping { //nolint:gocognit,cyclop
 				return fmt.Errorf("unknown page event: %q", event)
 			}
 
+			if event == common.EventPageMetricCalled {
+				// Register a custom regex function for the metric event
+				// that will be used to check URLs against the patterns.
+				// This is needed because we want to use the JavaScript regex
+				// to comply with what users expect when using the `tag` method.
+				_, err := rt.RunString(`
+					function _k6BrowserCheckRegEx(pattern, url) {
+						let r = pattern;
+						if (typeof pattern === 'string') {
+							r = new RegExp(pattern);
+						}
+						return r.test(url);
+					}
+				`)
+				if err != nil {
+					return fmt.Errorf("evaluating regex function: %w", err)
+				}
+			}
+
```

Comment:
We can later on find a better place to put this. Other metrics may need such an initialization step as well.

Reply:
Yeah, fair enough. Happy with the change. Agree that we can find a better place when the need arises 👍